plugins/modules/cloud/misc/terraform.py

```diff
@@ -449,12 +474,41 @@ def main():
     if state == 'present' and module.params.get('parallelism') is not None:
         command.append('-parallelism=%d' % module.params.get('parallelism'))
 
+    def process_complex_args(vars):
+        ret_out = []
+        if isinstance(vars, dict):
+            for k, v in vars.items():
+                if isinstance(v, dict):
+                    ret_out.append('{0}={{{1}}}'.format(k, process_complex_args(v)))
+                if isinstance(v, list):
+                    ret_out.append("{0}=[{1}]".format(k, process_complex_args(v)))
+                if isinstance(v, (integer_types, float, str, bool)):
+                    ret_out.append('{0}={1}'.format(k, json.dumps(v)))
```

Comment:
Why not use `elif` and a final `else` here as well?

Reply:
These are mutually exclusive checks, I didn't see the need for that.